datafusion/src/optimizer/simplify_expressions.rs

```diff
@@ -785,180 +786,166 @@ mod tests {
-        let expr_b = col("c").and(lit(true));
-        let expected = col("c");
+    fn test_simplify_and_true() {
+        let expr_a = lit(true).and(col("c2"));
+        let expr_b = col("c2").and(lit(true));
+        let expected = col("c2");
 
         assert_eq!(simplify(&expr_a), expected);
         assert_eq!(simplify(&expr_b), expected);
-        Ok(())
     }
 
     #[test]
-    fn test_simplify_multiply_by_one() -> Result<()> {
-        let expr_a = binary_expr(col("c"), Operator::Multiply, lit(1));
-        let expr_b = binary_expr(lit(1), Operator::Multiply, col("c"));
-        let expected = col("c");
+    fn test_simplify_multiply_by_one() {
+        let expr_a = binary_expr(col("c2"), Operator::Multiply, lit(1));
+        let expr_b = binary_expr(lit(1), Operator::Multiply, col("c2"));
+        let expected = col("c2");
 
         assert_eq!(simplify(&expr_a), expected);
         assert_eq!(simplify(&expr_b), expected);
-        Ok(())
     }
 
     #[test]
-    fn test_simplify_divide_by_one() -> Result<()> {
-        let expr = binary_expr(col("c"), Operator::Divide, lit(1));
-        let expected = col("c");
+    fn test_simplify_divide_by_one() {
+        let expr = binary_expr(col("c2"), Operator::Divide, lit(1));
+        let expected = col("c2");
 
         assert_eq!(simplify(&expr), expected);
-        Ok(())
     }
 
     #[test]
-    fn test_simplify_divide_by_same() -> Result<()> {
-        let expr = binary_expr(col("c"), Operator::Divide, col("c"));
+    fn test_simplify_divide_by_same() {
+        let expr = binary_expr(col("c2"), Operator::Divide, col("c2"));
         let expected = lit(1);
 
         assert_eq!(simplify(&expr), expected);
-        Ok(())
     }
 
     #[test]
-    fn test_simplify_simple_and() -> Result<()> {
+    fn test_simplify_simple_and() {
         // (c > 5) AND (c > 5)
-        let expr = (col("c").gt(lit(5))).and(col("c").gt(lit(5)));
-        let expected = col("c").gt(lit(5));
+        let expr = (col("c2").gt(lit(5))).and(col("c2").gt(lit(5)));
+        let expected = col("c2").gt(lit(5));
 
         assert_eq!(simplify(&expr), expected);
-        Ok(())
     }
 
     #[test]
-    fn test_simplify_composed_and() -> Result<()> {
+    fn test_simplify_composed_and() {
         // ((c > 5) AND (d < 6)) AND (c > 5)
         let expr = binary_expr(
-            binary_expr(col("c").gt(lit(5)), Operator::And, col("d").lt(lit(6))),
+            binary_expr(col("c2").gt(lit(5)), Operator::And, col("d").lt(lit(6))),
             Operator::And,
-            col("c").gt(lit(5)),
+            col("c2").gt(lit(5)),
         );
         let expected =
-            binary_expr(col("c").gt(lit(5)), Operator::And, col("d").lt(lit(6)));
+            binary_expr(col("c2").gt(lit(5)), Operator::And, col("d").lt(lit(6)));
 
         assert_eq!(simplify(&expr), expected);
-        Ok(())
     }
 
     #[test]
-    fn test_simplify_negated_and() -> Result<()> {
+    fn test_simplify_negated_and() {
```

Comment:
Why can't be simplified to `false`?

Reply:
There is no theoretical reason. I think this test is to ensure a particular corner case of how the rewrite rule for `A and (B and A)` is implemented.

Reply:
Tracking in https://github.com/apache/arrow-datafusion/issues/1406